packages/swapper/src/swappers/ButterSwap/swapperApi/checkTradeStatus.ts

```diff
@@ -13,24 +13,55 @@ const BUTTER_SWAP_STATES = (() => ({
   Failed: 6,
 }))()
 
+// Cache bridge IDs to avoid repeated API calls during polling
+// We use a module-level cache instead of storing in swap.metadata because:
+// 1. The bridge ID is only available after transaction submission (not at quote time like chainflipSwapId)
+// 2. checkTradeStatus doesn't have a mechanism to update swap state during polling
+// 3. This approach is simpler and doesn't require changes to core types or polling logic
+const bridgeIdCache = new Map<string, number>()
+
 export const checkTradeStatus = async (input: CheckTradeStatusInput): Promise<TradeStatus> => {
   const { txHash } = input
   try {
-    const infoResult = await getBridgeInfoBySourceHash(txHash)
-    if (infoResult.isErr()) {
+    let bridgeId = bridgeIdCache.get(txHash)
+
+    // Only fetch bridge info by source hash if we don't have the bridge ID cached
+    if (!bridgeId) {
+      const infoResult = await getBridgeInfoBySourceHash(txHash)
+      if (infoResult.isErr()) {
+        return {
+          status: TxStatus.Unknown,
+          buyTxHash: undefined,
+          message: infoResult.unwrapErr().message,
+        }
+      }
+      const basicInfo = infoResult.unwrap()
+      if (!basicInfo || !basicInfo.id) {
+        return { status: TxStatus.Unknown, buyTxHash: undefined, message: undefined }
+      }
+
+      // Cache the bridge ID for future calls
+      bridgeId = basicInfo.id
+      bridgeIdCache.set(txHash, bridgeId)
+    }
+
+    // Get the detailed bridge info using the cached ID
+    const detailedInfoResult = await getBridgeInfoById(bridgeId)
+    if (detailedInfoResult.isErr()) {
       return {
         status: TxStatus.Unknown,
         buyTxHash: undefined,
-        message: infoResult.unwrapErr().message,
+        message: detailedInfoResult.unwrapErr().message,
       }
     }
-    const info = infoResult.unwrap()
-    if (!info) {
+    const detailedInfo = detailedInfoResult.unwrap()
+    if (!detailedInfo) {
       return { status: TxStatus.Unknown, buyTxHash: undefined, message: undefined }
     }
+
     // State mapping: 0 = pending, 1 = complete, 6 = refunded/failed
     let status: TxStatus = (() => {
-      switch (info.state) {
+      switch (detailedInfo.state) {
         case BUTTER_SWAP_STATES.Pending:
           return TxStatus.Pending
         case BUTTER_SWAP_STATES.Confirmed:
```

Comment:
Didn't spot this one when reviewing initially since I didn't test swaps to longer outbound chains (namely, BTC, that's the only one like that Butter supports!), but now that I tested it in this PR, and especially with the addition of multiple Txs, we can improve things by checking if we have a `toHash` and returning `Confirmed` if so, since we're optimistic on UTXO Txs and balance will update after they hit the mempool (which they do the moment there is a broadcasted `toHash`)